extractor/extractor.go

```diff
@@ -0,0 +1,266 @@
+package extractor
+
+import (
+	"bufio"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/vision-cli/common/cases"
+	"github.com/vision-cli/common/transpiler/model"
+	protoparser "github.com/yoheimuta/go-protoparser"
+	parserStructs "github.com/yoheimuta/go-protoparser/parser"
+)
+
+func ProjectStructure(projectDirectory string) ([]model.Module, error) {
+	targetDir := filepath.Join(projectDirectory, "services")
+
+	// Get module names and api versions
+	modules, err := Modules(targetDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading modules: %w", err)
+	}
+
+	// Get services for each module
+	for i, module := range modules {
+		moduleNameWithVersion := fmt.Sprintf("%s.%s", module.Name, module.ApiVersion)
+		moduleDirectory := filepath.Join(targetDir, moduleNameWithVersion)
+		modules[i].Services, err = Services(moduleDirectory, &module)
+		if err != nil {
+			return nil, fmt.Errorf("error reading services: %w", err)
+		}
+	}
+
+	return modules, nil
+}
+
+func Modules(targetDir string) ([]model.Module, error) {
+	moduleDirs, err := os.ReadDir(targetDir)
```

Comment:
Pls use the file module in common to access the os. Reason is we can mock file and hence we can write tests without needing to access os. Pls see other tests in common that use file